### DISCUSSION TITLE
checkServerIdentity

### DISCUSSION THREAD

**Discussion by S222em** [nodejs/undici#1351] [Q&A] [answered]
Hello,

I am looking for something similar to `new Agent({ checkServerIdenty: () => undefined )}`. The request I am making is a `HTTPS` request, but the `hostname` will never match what's on the CA, as I am working with smart home devices. I have looked at `Agent#connect` but I cant find anything similar there.

**Comment by mcollina** [comment]
This is likely not supported, would you like to send a PR?

**Reply by S222em** [reply]
I took a look around, and the `buildConnector` function already supports `checkServerIdentity`, it's just not in the typings. By just adding `checkServerIdentity: () => undefined` with a `@ts-ignore` it works fine. I will just make a PR to add this to the typings.

```js
socket = tls.connect({
        highWaterMark: 16384, // TLS in node can't have bigger HWM anyway...
        ...options,
        servername,
        session,
        port: port || 443,
        host: hostname,
        // checkServerIdentity: () => undefined,
      })
```
I am not sure why this is not included in the tls typings from NodeJS

**Comment by S222em** [comment, marked as answer]
#1362
